### PR TITLE
Feat/(APPT-STQA-123): update schedule v2

### PR DESCRIPTION
**Pull request by GSwizzy**

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java [hunk 1]
```diff
@@ -44,7 +45,7 @@
  */
 @Repository
 public class JdbcVetRepositoryImpl implements VetRepository {
-
+   // JdbcTemplate jdbcTemplatee;
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java]: Delete if unused

File: src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java [hunk 2]
```diff
@@ -106,5 +107,31 @@ public Integer mapRow(ResultSet rs, int row) throws SQLException {
         return vets;
     }
 
+    @Override
+    public Vet findById(int id) {
+        List<Vet> vets = new ArrayList<>();
+        // Retrieve the list of all vets.
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java]: Comment needs update

File: src/main/java/org/springframework/samples/petclinic/web/VetController.java [hunk 3]
```diff
@@ -154,6 +141,57 @@ else if(selectedVet.getSpecialties().toString().contains("radiology"))
...
+        String[] weekDays = new String[]{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+        for(int i=0; i<schedules.size(); i++){
+            int id = Integer.parseInt(schedules.get(i));
+            scheduleList.add(new Schedule(id, weekDays[id - 1]));
+        }
+
+        vet.setScheduleInternal(scheduleList);
+
+        this.clinicService.saveVet(vet);
+        Thread.sleep(60000);
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/web/VetController.java]: Wait what?

File: src/main/webapp/WEB-INF/jsp/vets/modifySchedule.jsp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
+<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
+<%@ taglib prefix="th" uri="http://www.springframework.org/tags/form" %>
+
+
+<petclinic:layout pageName="vets">
+
+    <div>
+        <h1>${vet.firstName}'s Weekly Schedule</h1>
+<%--        <form:form action="/modifySchedule" modelAttribute="vet" class="form-horizontal" id="edit-vet-form" th:object="${vet}" method="POST">--%>
```
Comment (sylveon) [inline src/main/webapp/WEB-INF/jsp/vets/modifySchedule.jsp]: Remove all that commented out code completely

File: src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java [hunk 5]
```diff
@@ -378,21 +378,13 @@ void shouldDeleteVisitsByIdIn() {
...
-//    }
-
-
-//    @Test
-//    void shouldFindScheduleWithCorrectId() {
-//        Schedule sched6 = this.clinicService.findScheduleByVetId(6);
-//        assertThat(sched6.getVetId()).isEqualTo(6);
-//        assertThat(sched6.getDayAvailable()).isEqualTo(5);
+//@Test
+//void shouldFindVetById(){
```
Comment (sylveon) [inline src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java]: I would leave that uncommented

File: src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java [hunk 6]
```diff
@@ -106,5 +107,31 @@ public Integer mapRow(ResultSet rs, int row) throws SQLException {
...
+            BeanPropertyRowMapper.newInstance(Specialty.class));
+
+        // Build each vet's list of specialties.
+        fillSpecialties(vets);
+        return vets.get(0);
+    }
+
+    @Override
+    public void save(Vet vet) {
+//        BeanPropertySqlParameterSource parameterSource = new BeanPropertySqlParameterSource(vet);
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java]: Look at https://github.com/cgerard321/STQA-pet-clinic-F2020/blob/master/src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcPetRepositoryImpl.java#L89 for an implementation

Also add unit tests.

**Review by sylveon — CHANGES_REQUESTED** [review]
[CodeReviewCheckList (2).xlsx](https://github.com/cgerard321/STQA-pet-clinic-F2020/files/5657364/CodeReviewCheckList.2.xlsx)

**Review by Colorin09 — APPROVED** [review]

**Comment by Colorin09** [comment]
[CodeReviewCheckList - 2.xlsx](https://github.com/cgerard321/STQA-pet-clinic-F2020/files/5660178/CodeReviewCheckList.-.2.xlsx)

**Review by sylveon — APPROVED** [review]

**Pull request merged**